File: embeds/messageModeration.ts
import { Message, EmbedBuilder } from "discord.js";
import { ACTIONS, ATTR_PRETTY, Reason } from "../clients/constants.js";

export default function embedMessageModeration(message: Message, action: number, reasons: Reason[]) {
    const embed = new EmbedBuilder()
        .setTitle(`Flagged Message from ${message.author.username}`)
        .setURL(message.url)
        .setAuthor({ name: "Sibyl System", iconURL: message.client.user!.avatarURL()! })
        .setTimestamp()
        .setFooter({ text: message.id });

    if (ACTIONS[action] === "BAN") embed.setDescription(`Banned <@${message.author.id}>`);
    else if (ACTIONS[action] === "KICK") embed.setDescription(`Kicked <@${message.author.id}>`);
    else if (ACTIONS[action] === "MUTE") embed.setDescription(`Muted <@${message.author.id}>`);
    else if (ACTIONS[action] === "REMOVE") embed.setDescription("Removed Message");
    else if (ACTIONS[action] === "NOTIFY") embed.setDescription("Notified Moderators");

    for (const reason of reasons) {
        embed.addFields(
            { name: "\u200B", value: "\u200B" },
            { name: ATTR_PRETTY[reason.attribute as keyof typeof ATTR_PRETTY], value: reason.score.toString(), inline: true },
            { name: "Threshold", value: reason.threshold.toString(), inline: true }
        );
    }

    return embed;
}

File: embeds/message_moderation.py
from datetime import datetime, UTC
from discord import Embed, EmbedAuthor, EmbedFooter, Message
from clients.constants import ACTION, ATTR_PRETTY


def embed_message_moderation(
    message: Message, action: ACTION, reasons: list[tuple]
) -> Embed:
    embed = Embed(
        title=f"flagged message from {message.author.name}",
        url=message.jump_url,
        author=EmbedAuthor(name="sibylmod", icon_url=message.author.avatar.url),
        timestamp=datetime.now(UTC),
        footer=EmbedFooter(text=message.id),
    )

    if action is ACTION.BAN:
        embed.description = f"banned {message.author.mention}"
    elif action is ACTION.KICK:
        embed.description = f"kicked {message.author.mention}"
    elif action is ACTION.MUTE:
        embed.description = f"muted {message.author.mention}"
    elif action is ACTION.REMOVE:
        embed.description = "removed message"
    elif action is ACTION.NOTIFY:
        embed.description = "notified moderators"

    for reason in reasons:
        attribute, score, threshold = reason
        embed.add_field(name="\u200B", value="\u200B")
        embed.add_field(name=ATTR_PRETTY[attribute], value=score, inline=True)
        embed.add_field(name="threshold", value=threshold, inline=True)

    return embed
